url43a0865bb0_kevinjdolan_darwin_tgz-pJ8-tui_DriverJ8/src/darwInvest/news/utility/Stemmer.java

```diff
@@ -0,0 +1,562 @@
+			else if (doublec(k)) {  
+				k--;
+				int ch = b[k];
+				if (ch == 'l' || ch == 's' || ch == 'z')
+					k++;
+			}
+			else if (m() == 1 && cvc(k))
+				setto("e");
+		}
+	}
+	
+	/* step2() turns terminal y to i when there is another vowel in the stem. */
+	
+	private final void step2() {
+		if (ends("y") && vowelinstem()) 
+			b[k] = 'i'; 
+	}
+	
+	/* step3() maps double suffices to single ones. so -ization ( = -ize plus
+	      -ation) maps to -ize etc. note that the string before the suffix must give
+	      m() > 0. */
+	
+	private final void step3() { 
+		if (k == 0) return; /* For Bug 1 */ 
+		switch (b[k-1])
+		{
+			case 'a':
+				if (ends("ational")) { 
+					r("ate"); 
+					break; 
+				}
+				if (ends("tional")) { 
+					r("tion"); 
+					break; 
+				}
+				break;
+			case 'c':
+				if (ends("enci")) {
+					r("ence"); 
+					break; 
+				}
+				if (ends("anci")) { 
+					r("ance");
+					break; 
+				}
+				break;
+			case 'e':
+				if (ends("izer")) { 
+					r("ize"); 
+					break; 
+				}
+				break;
+			case 'l':
+				if (ends("bli")) { 
+					r("ble"); 
+					break; 
+				}
+				if (ends("alli")) {
+					r("al"); 
+					break; 
+				}
+				if (ends("entli")) {
+					r("ent"); 
+					break; 
+				}
+				if (ends("eli")) { 
+					r("e"); 
+					break; 
+				}
+				if (ends("ousli")) { 
+					r("ous"); 
+					break; 
+				}
+				break;
+			case 'o': 
+				if (ends("ization")) {
+					r("ize"); 
+					break; 
+				}
+				if (ends("ation")) {
+					r("ate"); 
+					break; 
+				}
+				if (ends("ator")) {
+					r("ate"); 
+					break; 
+				}
+				break;
+			case 's': 
+				if (ends("alism")) { 
+					r("al"); 
+					break; 
+				}
+				if (ends("iveness")) { 
+					r("ive"); 
+					break; 
+				}
+				if (ends("fulness")) { 
+					r("ful"); 
+					break; 
+				}
+				if (ends("ousness")) { 
+					r("ous"); 
+					break; 
+				}
+				break;
+			case 't': 
+				if (ends("aliti")) {
+					r("al"); 
+					break; 
+				}
+				if (ends("iviti")) { 
+					r("ive"); 
+					break; 
+				}
+				if (ends("biliti")) { 
+					r("ble"); 
+					break; 
+				}
+				break;
+			case 'g': 
+				if (ends("logi")) { 
+					r("log"); 
+					break; 
+				}
+		} 
+	}
+	
+	/* step4() deals with -ic-, -full, -ness etc. similar strategy to step3. */
+	private final void step4() { 
+		switch (b[k]) {
+			case 'e':
+				if (ends("icate")) { 
+					r("ic"); 
+					break; 
+				}
+				if (ends("ative")) { 
+					r(""); 
+					break; 
+				}
+				if (ends("alize")) { 
+					r("al"); 
+					break; 
+				}
+				break;
+			case 'i': 
+				if (ends("iciti")) { 
+					r("ic"); 
+					break; 
+				}
+				break;
+			case 'l': 
+				if (ends("ical")) { 
+					r("ic"); 
+					break; 
+				}
+				if (ends("ful")) { 
+					r(""); 
+					break; 
+				}
+				break;
+			case 's': 
+				if (ends("ness")) { 
+					r(""); 
+					break; 
+				}
+				break;
+		} 
+	}
+	
+	/* step5() takes off -ant, -ence etc., in context <c>vcvc<v>. */
+	
+	private final void step5() {   
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 26. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.